src/widgets/favoritewidget.cpp

```diff
@@ -50,12 +50,16 @@ QSize FavoriteWidget::sizeHint() const {
 }
 
 void FavoriteWidget::paintEvent(QPaintEvent* e) {
-  QStylePainter p(this);
+  // ignore text descent when vertically centering
+  QFontMetrics fontMetrics = QFontMetrics(this->font());
+  QRect rect = this->rect();
+  rect.setBottom(rect.bottom() - fontMetrics.descent());
```

Comment:
Hmm.. you got a point. what about handling this in resize event like this ?
void FavoriteWidget::resizeEvent(QResizeEvent *) {
  rect_ = this->rect();
  rect_.setBottom(rect_.bottom() - QFontMetrics(this->font()).descent());
}